src/evergreen/performance_results.py

```diff
@@ -0,0 +1,177 @@
+# -*- encoding: utf-8 -*-
+"""Performance results representation of evergreen."""
+from __future__ import absolute_import
+
+import copy
+
+from evergreen.base import _BaseEvergreenObject, evg_attrib, evg_short_datetime_attrib, \
+    evg_datetime_attrib
+from evergreen.util import parse_evergreen_datetime
+
+
+class TestResult(_BaseEvergreenObject):
+    """Representation of a test result from Evergreen."""
+    thread_level = evg_attrib('thread_level')
+    operations_per_second = evg_attrib('ops_per_sec_values')
+    mean_operations_per_second = evg_attrib('ops_per_sec')
+
+    def __init__(self, json, api):
+        """Create an instance of a test result."""
+        super(TestResult, self).__init__(json, api)
+
+
+class TestRun(_BaseEvergreenObject):
+    """Representation of a test run from Evergreen."""
+    workload = evg_attrib('workload', default='microbenchmarks')
+    test_name = evg_attrib('name')
+
+    def __init__(self, test_result, api, parent):
+        """Create an instance of a test run."""
+        super(TestRun, self).__init__(test_result, api)
+        if parent:
+            self.project_id = parent.project_id
+            self.task_name = parent.task_name
+            self.task_id = parent.task_id
+            self.variant = parent.variant
+            self.version_id = parent.version_id
+            self.revision = parent.revision
+            self.order = parent.order
+            self.create_time = parent.create_time
+
+        # Microbenchmarks stores the 'start' and 'end' time of the test in the inner 'results' field
+        # while sys-perf stores it in the outer 'results' field.
+        self.start = parse_evergreen_datetime(
+            test_result['start'] if 'start' in test_result else test_result['results'][
```

Comment:
You might be able to use `get` to make this a little cleaner.

```
test_result.get('start', test_result['results']['start'])
```

same one the next line for `self.end`.